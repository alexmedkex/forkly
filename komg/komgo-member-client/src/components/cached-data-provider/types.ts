export interface ICachedData<T> {
  cached?: T
}

export interface ICacheOwnProps<T> {
  id: string
  data?: T
  children: (props: ICachedData<T>) => React.ReactElement
}
