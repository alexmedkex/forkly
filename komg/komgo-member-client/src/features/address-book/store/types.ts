import { List } from 'immutable'
import { ImmutableMap } from '../../../utils/types'
import { Action } from 'redux'
import { IMember } from '../../members/store/types'

export enum AddressBookActionType {
  GET_COMPANIES_REQUEST = '@@addressBook/GET_COMPANIESS_REQUEST',
  GET_COMPANIES_SUCCESS = '@@addressBook/GET_COMPANIESS_SUCCESS',
  GET_COMPANIES_FAILURE = '@@addressBook/GET_COMPANIESS_FAILURE',
  GET_COMPANY_REQUEST = '@@addressBook/GET_COMPANY_REQUEST',
  GET_COMPANY_SUCCESS = '@@addressBook/GET_COMPANY_SUCCESS',
  GET_COMPANY_FAILURE = '@@addressBook/GET_COMPANY_FAILURE',
  GENERATE_MEMBER_PACKAGE_REQUEST = '@@addressBook/GENERATE_MEMBER_PACKAGE_REQUEST',
  GENERATE_MEMBER_PACKAGE_SUCCESS = '@@addressBook/GENERATE_MEMBER_PACKAGE_SUCCESS',
  GENERATE_MEMBER_PACKAGE_FAILURE = '@@addressBook/GENERATE_MEMBER_PACKAGE_FAILURE',
  ADD_COMPANY_TO_ENS_REQUEST = '@@addressBook/ADD_COMPANY_TO_ENS_REQUEST',
  ADD_COMPANY_TO_ENS_SUCCESS = '@@addressBook/ADD_COMPANY_TO_ENS_SUCCESS',
  ADD_COMPANY_TO_ENS_FAILURE = '@@addressBook/ADD_COMPANY_TO_ENS_FAILURE',
  CONFIGURE_MQ_REQUEST = '@@addressBook/CONFIGURE_MQ_REQUEST',
  CONFIGURE_MQ_SUCCESS = '@@addressBook/CONFIGURE_MQ_SUCCESS',
  CONFIGURE_MQ_FAILURE = '@@addressBook/CONFIGURE_MQ_FAILURE',
  CREATE_COMPANY_REQUEST = '@@addressBook/CREATE_COMPANY_REQUEST',
  CREATE_COMPANY_SUCCESS = '@@addressBook/CREATE_COMPANY_SUCCESS',
  CREATE_COMPANY_FAILURE = '@@addressBook/CREATE_COMPANY_FAILURE',
  UPDATE_COMPANY_REQUEST = '@@addressBook/UPDATE_COMPANY_REQUEST',
  UPDATE_COMPANY_SUCCESS = '@@addressBook/UPDATE_COMPANY_SUCCESS',
  UPDATE_COMPANY_FAILURE = '@@addressBook/UPDATE_COMPANY_FAILURE',
  DeleteCompanyByIdSuccess = '@@addressBook/DELETE_COMPANY_BY_ID_SUCCESS'
}

export interface AddressBookStateProperties {
  companies: List<IMember>
}

export type AddressBookState = ImmutableMap<AddressBookStateProperties>

export interface GetCompaniesRequest extends Action {
  type: AddressBookActionType.GET_COMPANIES_SUCCESS
  payload: IMember[]
}
export interface GetCompanyRequest extends Action {
  type: AddressBookActionType.GET_COMPANY_SUCCESS
  payload: IMember
}
export interface GenerateMemberPackageRequest extends Action {
  type: AddressBookActionType.GENERATE_MEMBER_PACKAGE_SUCCESS
  payload: IMember
}
export interface AddCompanyToENSRequest extends Action {
  type: AddressBookActionType.ADD_COMPANY_TO_ENS_SUCCESS
  payload: IMember
}
export interface ConfigureMQRequest extends Action {
  type: AddressBookActionType.CONFIGURE_MQ_SUCCESS
  payload: IMember
}

export interface CreateCompanyRequest extends Action {
  type: AddressBookActionType.CREATE_COMPANY_SUCCESS
  payload: IMember
}
export interface UpdateCompanyRequest extends Action {
  type: AddressBookActionType.UPDATE_COMPANY_SUCCESS
  payload: IMember
}

export interface DeleteCompanyByIdSuccess extends Action {
  type: AddressBookActionType.DeleteCompanyByIdSuccess
  payload: { id: string }
}

export type AddressBookActions =
  | GetCompaniesRequest
  | GetCompanyRequest
  | CreateCompanyRequest
  | UpdateCompanyRequest
  | GenerateMemberPackageRequest
  | AddCompanyToENSRequest
  | ConfigureMQRequest
  | DeleteCompanyByIdSuccess
