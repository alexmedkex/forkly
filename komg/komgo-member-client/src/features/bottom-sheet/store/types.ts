import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import {
  CreateDocumentError,
  CreateDocumentSuccess,
  DocumentRegisteredError,
  DocumentRegisteredSuccess,
  HasId,
  HasName
} from '../../document-management/store/types'
import { ActionCreator } from 'react-redux'

export enum BottomSheetActionType {
  REMOVE_BOTTOMSHEET_ITEM = '@@btsh/REMOVE_BOTTOMSHEET_ITEM',
  RETRY_BOTTOMSHEET_ITEM = '@@btsh/RETRY_BOTTOMSHEET_ITEM',
  UPDATE_BOTTOMSHEET_ITEM = '@@btsh/UPDATE_BOTTOMSHEET_ITEM',
  CREATE_BOTTOMSHEET_ITEM = '@@btsh/CREATE_BOTTOMSHEET_ITEM'
}

export interface BottomSheetItem extends HasId, HasName {
  state: BottomSheetStatus
  itemType?: BottomSheetItemType
  displayStatus?: string
  action?: ActionCreator<any>
  navigationLink?: string
}

export enum BottomSheetItemType {
  REGISTER_KYC_DOCUMENT,
  CREATE_MAGIC_LINK,
  REGISTER_RD_DOCUMENT,
  REGISTER_TRADE_DOCUMENT
}

export enum BottomSheetStatus {
  NEW = 'NEW',
  PENDING = 'PENDING',
  REGISTERED = 'REGISTERED',
  FAILED = 'FAILED'
}

export interface BottomSheetStateFields {
  visible: boolean
  items: BottomSheetItem[]
}

export interface RemoveBottomSheetItem extends Action {
  type: BottomSheetActionType.REMOVE_BOTTOMSHEET_ITEM
  payload: HasId
}

export interface RetryRegisterItem extends Action {
  type: BottomSheetActionType.RETRY_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface UpdateBottomSheetItem extends Action {
  type: BottomSheetActionType.UPDATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export interface CreateBottomSheetItem extends Action {
  type: BottomSheetActionType.CREATE_BOTTOMSHEET_ITEM
  payload: BottomSheetItem
}

export type BottomSheetState = ImmutableMap<BottomSheetStateFields>

export type BottomSheetAction =
  | DocumentRegisteredSuccess
  | DocumentRegisteredError
  | CreateDocumentSuccess
  | CreateDocumentError
  | RemoveBottomSheetItem
  | RetryRegisterItem
  | UpdateBottomSheetItem
  | CreateBottomSheetItem
