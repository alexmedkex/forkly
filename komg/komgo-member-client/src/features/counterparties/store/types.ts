import { Action } from 'redux'
import { ImmutableMap } from '../../../utils/types'
import { IX500Name } from '@komgo/types'

// Actions types
export enum CounterpartiesActionType {
  FETCH_CONNECTED_COUNTERPARTIES_REQUEST = '@@counterparties/FETCH_CONNECTED_COUNTERPARTIES_REQUEST',
  FETCH_CONNECTED_COUNTERPARTIES_SUCCESS = '@@counterparties/FETCH_CONNECTED_COUNTERPARTIES_SUCCESS',
  FETCH_CONNECTED_COUNTERPARTIES_FAILURE = '@@counterparties/FETCH_CONNECTED_COUNTERPARTIES_FAILURE',
  FETCH_NOT_CONNECTED_COUNTERPARTIES_REQUEST = '@@counterparties/FETCH_NOT_CONNECTED_COUNTERPARTIES_REQUEST',
  FETCH_NOT_CONNECTED_COUNTERPARTIES_SUCCESS = '@@counterparties/FETCH_NOT_CONNECTED_COUNTERPARTIES_SUCCESS',
  FETCH_NOT_CONNECTED_COUNTERPARTIES_FAILURE = '@@counterparties/FETCH_NOT_CONNECTED_COUNTERPARTIES_FAILURE',
  ADD_COUNTERPARTY_REQUEST = '@@counterparties/ADD_COUNTERPARTY_REQUEST',
  ADD_COUNTERPARTY_SUCCESS = '@@counterparties/ADD_COUNTERPARTY_SUCCESS',
  ADD_COUNTERPARTY_FAILURE = '@@counterparties/ADD_COUNTERPARTY_FAILURE',
  FETCH_COUNTERPARTY_REQ_REQUEST = '@@counterparties/FETCH_COUNTERPARTY_REQ_REQUEST',
  FETCH_COUNTERPARTY_REQ_SUCCESS = '@@counterparties/FETCH_COUNTERPARTY_REQ_SUCCESS',
  FETCH_COUNTERPARTY_REQ_FAILURE = '@@counterparties/FETCH_COUNTERPARTY_REQ_FAILURE',
  RESEND_COUNTERPARTY_REQUEST = '@@counterparties/RESEND_COUNTERPARTY_REQUEST',
  RESEND_COUNTERPARTY_SUCCESS = '@@counterparties/RESEND_COUNTERPARTY_SUCCESS',
  RESEND_COUNTERPARTY_FAILURE = '@@counterparties/RESEND_COUNTERPARTY_FAILURE',
  RESPONSE_ON_COUNTERPARTY_REQ_REQUEST = '@@counterparties/RESPONSE_ON_COUNTERPARTY_REQ_REQUEST',
  RESPONSE_ON_COUNTERPARTY_REQ_SUCCESS = '@@counterparties/RESPONSE_ON_COUNTERPARTY_REQ_SUCCESS',
  RESPONSE_ON_COUNTERPARTY_REQ_FAILURE = '@@counterparties/RESPONSE_ON_COUNTERPARTY_REQ_FAILURE',
  SEARCH_COUNTERPARTY = '@@counterparties/SEARCH_COUNTERPARTY',
  SORT_CONNECTED_COUNTERPARTIES = '@@counterparties/SORT_CONNECTED_COUNTERPARTIES',
  SET_ADD_COUNTERPARTIES = '@@counterparties/SET_ADD_COUNTERPARTIES',
  SET_COUNTERPARTY_MODAL = '@@counterparties/SET_COUNTERPARTY_MODAL',
  SET_REQUEST_ACTION = '@@counterparties/SET_REQUEST_ACTION',
  FETCH_COUNTERPARTY_PROFILE_REQUEST = '@@counterparties/FETCH_COUNTERPARTY_PROFILE_REQUEST',
  FETCH_COUNTERPARTY_PROFILE_SUCCESS = '@@counterparties/FETCH_COUNTERPARTY_PROFILE_SUCCESS',
  FETCH_COUNTERPARTY_PROFILE_FAILURE = '@@counterparties/FETCH_COUNTERPARTY_PROFILE_FAILURE',
  UPDATE_COUNTERPARTY_PROFILE_REQUEST = '@@counterparties/UPDATE_COUNTERPARTY_PROFILE_REQUEST',
  UPDATE_COUNTERPARTY_PROFILE_SUCCESS = '@@counterparties/UPDATE_COUNTERPARTY_PROFILE_SUCCESS',
  UPDATE_COUNTERPARTY_PROFILE_FAILURE = '@@counterparties/UPDATE_COUNTERPARTY_PROFILE_FAILURE',
  CREATE_COUNTERPARTY_PROFILE_REQUEST = '@@counterparties/CREATE_COUNTERPARTY_PROFILE_REQUEST',
  CREATE_COUNTERPARTY_PROFILE_SUCCESS = '@@counterparties/CREATE_COUNTERPARTY_PROFILE_SUCCESS',
  CREATE_COUNTERPARTY_PROFILE_FAILURE = '@@counterparties/CREATE_COUNTERPARTY_PROFILE_FAILURE'
}

// State
export interface CounterpartiesStateFields {
  counterpartiesSearch: string
  counterparties: Counterparty[]
  counterpartiesSort: Sort
  notConnectedCounterpartySearch: string
  notConnectedCounterparties: NotConnectedCounterparty[]
  addCounterparties: string[]
  isAddModalOpen: boolean
  requestResponseActionStatus: boolean
  counterpartyRequest: CounterpartyRequest | null
  counterpartyProfiles: Map<string, CounterpartyProfile>
}

// Immutable state
export type CounterpartiesState = ImmutableMap<CounterpartiesStateFields>

export enum CouneterpartyStatus {
  COMPLETED = 'COMPLETED',
  PENDING = 'PENDING',
  WAITING = 'WAITING'
}

export const CouneterpartyStatusText: { [key: string]: string } = {
  [CouneterpartyStatus.COMPLETED]: 'Completed',
  [CouneterpartyStatus.PENDING]: 'Request pending',
  [CouneterpartyStatus.WAITING]: 'Action Required'
}

// Data types
export interface Counterparty {
  isFinancialInstitution: boolean
  staticId: string
  isMember: boolean
  x500Name: IX500Name
  covered: boolean
  status?: string
  timestamp?: Date
  profile?: CounterpartyProfile
}

export interface CounterpartyRequest extends Counterparty {
  requestId: string
}

export interface CompanyInfo {
  CN: string
  O: string
  C: string
  L: string
  STREET: string
  PC: string
}

export interface NotConnectedCounterparty {
  staticId: string
  isMember: boolean
  x500Name: CompanyInfo
  status?: CouneterpartyStatus
}

export interface SearchCounterpartyPayload {
  search: string
  typeCounterparty: 'counterpartiesSearch' | 'notConnectedCounterpartySearch'
}

export interface Sort {
  column: string
  order: 'ascending' | 'descending' | ''
}

export interface ModalPayload {
  name: 'isAddModalOpen'
  value: boolean
}

export interface CounterpartyProfile {
  id: string
  counterpartyId: string
  riskLevel: RiskLevel
  renewalDate: string
  managedById: string
}

export enum RiskLevel {
  unspecified = '',
  low = 'low',
  medium = 'medium',
  high = 'high'
}

// Actions
export interface FetchCounterpartiesSuccess extends Action {
  payload: Counterparty[]
  type: CounterpartiesActionType.FETCH_CONNECTED_COUNTERPARTIES_SUCCESS
}

export interface FetchCounterpartiesError extends Action {
  type: CounterpartiesActionType.FETCH_CONNECTED_COUNTERPARTIES_FAILURE
  payload: string
}

export interface FetchNotConnectedCounterpartiesSuccess extends Action {
  payload: NotConnectedCounterparty[]
  type: CounterpartiesActionType.FETCH_NOT_CONNECTED_COUNTERPARTIES_SUCCESS
}

export interface FetchNotConnectedCounterpartiesError extends Action {
  type: CounterpartiesActionType.FETCH_NOT_CONNECTED_COUNTERPARTIES_FAILURE
  payload: string
}

export interface RequestFetchSuccess extends Action {
  type: CounterpartiesActionType.FETCH_COUNTERPARTY_REQ_SUCCESS
  payload: any
}

export interface SearchCounterparty extends Action {
  type: CounterpartiesActionType.SEARCH_COUNTERPARTY
  payload: SearchCounterpartyPayload
}

export interface SortConnectedCounterparties extends Action {
  type: CounterpartiesActionType.SORT_CONNECTED_COUNTERPARTIES
  payload: Sort
}

export interface AddCounterpartySuccess extends Action {
  type: CounterpartiesActionType.ADD_COUNTERPARTY_SUCCESS
  payload: string[]
}

export interface SetCounterpartyModal extends Action {
  type: CounterpartiesActionType.SET_COUNTERPARTY_MODAL
  payload: ModalPayload
}

export interface SetAddCounterparties extends Action {
  type: CounterpartiesActionType.SET_ADD_COUNTERPARTIES
  payload: string[]
}

export interface CounterPartyResponseAction extends Action {
  type: CounterpartiesActionType.SET_REQUEST_ACTION
  payload: {
    status: boolean
  }
}

export interface FetchCounterpartyProfileSuccess extends Action {
  type: CounterpartiesActionType.FETCH_COUNTERPARTY_PROFILE_SUCCESS
  payload: CounterpartyProfile
}

export interface FetchCounterpartyProfileFailure extends Action {
  type: CounterpartiesActionType.FETCH_COUNTERPARTY_PROFILE_FAILURE
  payload: {
    error: any
    counterpartyId: string
  }
}

export interface CreateCounterpartyProfileSuccess extends Action {
  type: CounterpartiesActionType.CREATE_COUNTERPARTY_PROFILE_SUCCESS
  payload: CounterpartyProfile
}

export interface UpdateCounterpartyProfileSuccess extends Action {
  type: CounterpartiesActionType.UPDATE_COUNTERPARTY_PROFILE_SUCCESS
  payload: CounterpartyProfile
}

// All actions
export type CounterpartiesAction =
  | FetchCounterpartiesSuccess
  | FetchCounterpartiesError
  | RequestFetchSuccess
  | SetCounterpartyModal
  | FetchNotConnectedCounterpartiesError
  | FetchNotConnectedCounterpartiesSuccess
  | SearchCounterparty
  | SortConnectedCounterparties
  | AddCounterpartySuccess
  | SetAddCounterparties
  | CounterPartyResponseAction
  | FetchCounterpartyProfileSuccess
  | FetchCounterpartyProfileFailure
  | CreateCounterpartyProfileSuccess
  | UpdateCounterpartyProfileSuccess
