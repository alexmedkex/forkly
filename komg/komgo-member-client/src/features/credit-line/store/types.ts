import { ImmutableMap } from '../../../utils/types'
import { Map } from 'immutable'
import {
  ICreditLine,
  IRiskCoverData,
  Currency,
  ICreditLineResponse,
  ISharedCreditLine,
  IProductContext,
  ICreditLineRequest,
  ICreateCreditLineRequest,
  IInformationShared
} from '@komgo/types'
import { Products } from '../../document-management/constants/Products'
import { SubProducts } from '../../document-management/constants/SubProducts'
import { IMember } from '../../members/store/types'

export enum CreditLineActionType {
  FetchCreditLinesRequest = '@@risk-cover/FETCH_CREDIT_LINES_REQUEST',
  FetchCreditLinesSuccess = '@@risk-cover/FETCH_CREDIT_LINES_SUCCESS',
  FetchCreditLinesFailure = '@@risk-cover/FETCH_CREDIT_LINES_FAILURE',
  GetCreditLineRequest = '@@risk-cover/GET_CREDIT_LINE_REQUEST',
  GetCreditLineSuccess = '@@risk-cover/GET_CREDIT_LINE_SUCCESS',
  GetCreditLineFailure = '@@risk-cover/GET_CREDIT_LINE_FAILURE',
  CreateCreditLineRequest = '@@risk-cover/CREATE_CREDIT_LINE_REQUEST',
  CreateCreditLineSuccess = '@@risk-cover/CREATE_CREDIT_LINE_SUCCESS',
  CreateCreditLineFailure = '@@risk-cover/CREATE_CREDIT_LINE_FAILURE',
  EditCreditLineRequest = '@@risk-cover/EDIT_CREDIT_LINE_REQUEST',
  EditCreditLineSuccess = '@@risk-cover/EDIT_CREDIT_LINE_SUCCESS',
  EditCreditLineFailure = '@@risk-cover/EDIT_CREDIT_LINE_FAILURE',
  FetchDisclosedCreditLineSummariesRequest = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINE_SUMMARIES_REQUEST',
  FetchDisclosedCreditLineSummariesSuccess = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINE_SUMMARIES_SUCCESS',
  FetchDisclosedCreditLineSummariesFailure = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINE_SUMMARIES_FAILURE',
  FetchDisclosedCreditLinesForCounterpartyRequest = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINES_FOR_COUNTERPARTY_REQUEST',
  FetchDisclosedCreditLinesForCounterpartySuccess = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINES_FOR_COUNTERPARTY_SUCCESS',
  FetchDisclosedCreditLinesForCounterpartyFailure = '@@risk-cover/FETCH_DISCLOSED_CREDIT_LINES_FOR_COUNTERPARTY_FAILURE',
  RemoveCreditLineRequest = '@@risk-cover/REMOVE_CREDIT_LINE_REQUEST',
  RemoveCreditLineSuccess = '@@risk-cover/REMOVE_CREDIT_LINE_SUCCESS',
  RemoveCreditLineFailure = '@@risk-cover/REMOVE_CREDIT_LINE_FAILURE',
  CreateReqInformationRequest = '@@risk-cover/CREATE_REQ_INFORMATION_REQUEST',
  CreateReqInformationSuccess = '@@risk-cover/CREATE_REQ_INFORMATION_SUCCESS',
  CreateReqInformationFailure = '@@risk-cover/CREATE_REQ_INFORMATION_FAILURE',
  FetchRequestsRequest = '@@risk-cover/FETCH_REQ_REQUEST',
  FetchRequestsSuccess = '@@risk-cover/FETCH_REQ_SUCCESS',
  FetchRequestsFailure = '@@risk-cover/FETCH_REQ_FAILURE',
  DeclineAllRequestsRequest = '@@risk-cover/DECLINE_ALL_REQ_REQUEST',
  DeclineAllRequestsSuccess = '@@risk-cover/DECLINE_ALL_REQ_SUCCESS',
  DeclineAllRequestsFailure = '@@risk-cover/DECLINE_ALL_REQ_FAILURE'
}

export enum CreditLineType {
  RiskCover = 'riskCover',
  BankLine = 'bankLine'
}

export interface IExtendedCreditLine extends ICreditLineResponse {
  counterpartyName: string
  counterpartyLocation: string
  data: IRiskCoverData
  sharedCreditLines: IExtendedSharedCreditLine[]
}

export interface IExtendedSharedCreditLine extends ISharedCreditLine<IInformationShared> {
  updatedAt?: string
  counterpartyName: string
}

export interface ICreateOrEditSharedCreditLine {
  counterpartyStaticId: string
  sharedWithStaticId: string
  data: IInformationShared
  requestStaticId?: string
  staticId?: string
  updatedAt?: string
}
export interface ICreateOrEditCreditLineForm {
  context: IProductContext
  counterpartyStaticId: string
  appetite: boolean
  availability: boolean
  creditLimit?: number
  availabilityAmount?: number
  data?: IRiskCoverData
  currency: Currency
  sharedCreditLines: ICreateOrEditSharedCreditLine[]
  availabilityAmountUpdatedAt?: string | Date
  staticId?: string
  creditExpiryDate: string | Date
}

export interface IProductProps {
  productId: Products
  subProductId: SubProducts
}

// TODO: move to komgo types
export interface IDisclosedCreditLineSummary {
  counterpartyStaticId: string
  lowestFee: number
  availabilityCount: number
  appetiteCount: number
  _id: string
}

export interface IExtendedCreditLineRequest extends ICreditLineRequest {
  counterpartyName: string
  companyName: string
}

export interface IDisclosedCreditLineSummaryEnriched extends IDisclosedCreditLineSummary {
  counterpartyName: string
  counterpartyLocation: string
}

export interface IDisclosedCreditLineEnriched extends IDisclosedCreditLine {
  companyName: string
  companyLocation: string
  counterpartyName: string
}

export interface IDisclosedCreditLine extends ICreditLine {
  ownerStaticId: string
  data?: any // TODO: move later to the right type inside `ICreditLine`
}

export interface IRequestInformationForm {
  mailTo: boolean
  comment: string
  requestForId: string
  companyIds: string[]
}

export interface IRequestCreditLineForm extends IRequestInformationForm {
  context: IProductContext
}

export interface IMailToData {
  email?: string
  subject?: string
  body?: string
}

export interface RiskCoverProperties {
  creditLinesById: Map<string, ICreditLineResponse>
  disclosedCreditLineSummariesById: Map<string, IDisclosedCreditLineSummary>
  disclosedCreditLinesById: Map<string, IDisclosedCreditLine>
  requestsById: Map<string, ICreditLineRequest>
}

export interface CreditLinesProperties {
  [CreditLineType.RiskCover]: ImmutableMap<RiskCoverProperties>
  [CreditLineType.BankLine]: ImmutableMap<RiskCoverProperties>
}

export type CreditLinesState = ImmutableMap<CreditLinesProperties>

export interface IMemberWithDisabledFlag extends IMember {
  disabled?: boolean
}
