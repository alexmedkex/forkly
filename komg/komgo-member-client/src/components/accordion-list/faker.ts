import { HasIdAndName } from './AccordionList'

const defaultHasIdAndName: HasIdAndName = { id: '-1', name: 'foo' }

export const fakeHasIdAndName = (param: Partial<HasIdAndName>): HasIdAndName => {
  return {
    id: param.id || defaultHasIdAndName.id,
    name: param.name || defaultHasIdAndName.name
  }
}
