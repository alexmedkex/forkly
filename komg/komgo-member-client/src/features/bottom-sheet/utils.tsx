import { BottomSheetItem, BottomSheetStatus, BottomSheetAction, BottomSheetItemType } from './store/types'
import React from 'react'

export const generateBottomsheetTitle = (items: BottomSheetItem[]) => {
  let textHeader = ''
  if (items[0]) {
    const latestStatus = items[0].state
    const count = items.filter(x => x.state === latestStatus).length
    textHeader = getPrintableStatus(latestStatus, count)
  }
  return textHeader
}

const getPrintableStatus = (latestStatus: BottomSheetStatus, count: number) => {
  switch (latestStatus) {
    case BottomSheetStatus.REGISTERED:
      return `${count} document${count > 1 ? 's' : ''} registered`
    case BottomSheetStatus.PENDING:
      return `${count} document${count > 1 ? 's' : ''} pending`
    case BottomSheetStatus.FAILED:
      return `${count} failed registration`
  }
}

export const getStatus = (action: BottomSheetAction): BottomSheetStatus => {
  return BottomSheetStatus.PENDING
}

export function fakeBottomsheetItem<T extends BottomSheetItem>(partial?: Partial<T>): BottomSheetItem {
  const { id, state, name, itemType } = {
    ...{
      id: '-1',
      name: 'anon',
      state: BottomSheetStatus.PENDING,
      itemType: BottomSheetItemType.REGISTER_KYC_DOCUMENT
    },
    ...partial
  }
  return { id, state, name, itemType }
}
