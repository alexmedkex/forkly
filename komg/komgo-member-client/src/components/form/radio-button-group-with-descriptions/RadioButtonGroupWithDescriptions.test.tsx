import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'
import { fakeFormik } from '../../../features/receivable-discounting-legacy/utils/faker'
import {
  RadioButtonGroupWithDescriptions,
  IRadioButtonGroupWithDescriptionsProps,
  StyledRadio,
  IRadioButtonWithDescription,
  StyledTitleContainer,
  StyledHeaderTitle
} from './RadioButtonGroupWithDescriptions'
import { RequestType } from '@komgo/types'
import { shallow } from 'enzyme'
import { displayRequestType } from '../../../features/receivable-discounting-legacy/utils/displaySelectors'
import { Strings } from '../../../features/receivable-discounting-legacy/resources/strings'

describe('RadioButtonGroupWithDescriptions', () => {
  let testProps: IRadioButtonGroupWithDescriptionsProps

  const radioButtonTestOptions: IRadioButtonWithDescription[] = [
    {
      value: RequestType.RiskCover,
      label: displayRequestType(RequestType.RiskCover),
      description: Strings.RiskCoverDescription,
      disabled: true
    },
    {
      value: RequestType.RiskCoverDiscounting,
      label: displayRequestType(RequestType.RiskCoverDiscounting),
      description: Strings.RiskCoverReceivableDiscountingDescription
    },
    {
      value: RequestType.Discount,
      label: displayRequestType(RequestType.Discount),
      description: Strings.ReceivableDiscountingDescription
    }
  ]

  beforeEach(() => {
    testProps = {
      groupTitle: 'Select an option',
      formik: fakeFormik,
      options: radioButtonTestOptions,
      name: 'requestType'
    }
  })

  it('matches snapshot ', () => {
    expect(render(<RadioButtonGroupWithDescriptions {...testProps} />).asFragment()).toMatchSnapshot()
  })

  it('leaves out the disabled option - leaving 2 Form Fields', () => {
    const wrapper = shallow(<RadioButtonGroupWithDescriptions {...testProps} />)

    expect(wrapper.find(StyledRadio).length).toBe(2)
  })

  it('shows a groupTitle if provided', () => {
    const wrapper = shallow(<RadioButtonGroupWithDescriptions {...testProps} />)

    expect(wrapper.find(StyledHeaderTitle).exists()).toBeTruthy()
  })

  it('does not show a groupTitle is not provided', () => {
    delete testProps.groupTitle

    const wrapper = shallow(<RadioButtonGroupWithDescriptions {...testProps} />)

    expect(wrapper.find(StyledHeaderTitle).exists()).toBeFalsy()
  })

  it('calls setFieldValue when option changed', async done => {
    const { getByTestId } = render(<RadioButtonGroupWithDescriptions {...testProps} />)

    const option1Radio = getByTestId('radioButtongroup-RISK_COVER_DISCOUNTING').querySelector('input')

    fireEvent.change(option1Radio, { target: { value: option1Radio.value } })
    fireEvent.blur(option1Radio)

    wait(() => {
      expect(testProps.formik.setFieldValue).toHaveBeenCalledWith('requestType', RequestType.RiskCoverDiscounting)
      done()
    })
  })
})
