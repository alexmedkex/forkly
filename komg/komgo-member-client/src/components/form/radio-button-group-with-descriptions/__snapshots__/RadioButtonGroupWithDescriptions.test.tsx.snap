// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonGroupWithDescriptions matches snapshot  1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 12px 0px 12px 10px;
}

.c1 {
  font-size: 21px;
  margin-bottom: 20px;
}

<div
    class="ui container c0"
  >
    <h3
      class="c1"
    >
      Select an option
    </h3>
  </div>
  .c0 {
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 5px;
}

.c0 > div {
  margin: 5px !important;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  padding: 16px;
  width: 200px;
  border: 1px solid #c0cfde;
  border-radius: 5px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c2 {
  padding-bottom: 10px;
}

.c2 > label {
  font-weight: bold;
  text-align: bottom;
}

.c3 {
  margin-left: 32px;
}

<div
    class="ui container c0"
  >
    <div
      class="field c1"
    >
      <div
        class="ui radio checkbox c2"
        data-test-id="radioButtongroup-RISK_COVER_DISCOUNTING"
      >
        <input
          class="hidden"
          id="radioButtongroup-RISK_COVER_DISCOUNTING"
          name="radioGroup"
          readonly=""
          tabindex="0"
          type="radio"
          value="RISK_COVER_DISCOUNTING"
        />
        <label
          for="radioButtongroup-RISK_COVER_DISCOUNTING"
        >
          Risk cover with discount option
        </label>
      </div>
      <p
        class="c3"
      >
        Cover the risk with the option to request discounting
      </p>
    </div>
    <div
      class="field c1"
    >
      <div
        class="ui radio checkbox c2"
        data-test-id="radioButtongroup-DISCOUNT"
      >
        <input
          class="hidden"
          id="radioButtongroup-DISCOUNT"
          name="radioGroup"
          readonly=""
          tabindex="0"
          type="radio"
          value="DISCOUNT"
        />
        <label
          for="radioButtongroup-DISCOUNT"
        >
          Discount
        </label>
      </div>
      <p
        class="c3"
      >
        Discount with or without recourse
      </p>
    </div>
  </div>
</DocumentFragment>
`;
